src/bootstrap/src/core/sanity.rs

```diff
@@ -177,6 +177,31 @@ than building it.
             continue;
         }
 
+        // Check if there exists a built-in target in the list of supported targets.
+        let mut has_target = false;
+        let target_str = target.to_string();
+
+        let supported_target_list = Command::new("rustc")
+            .current_dir(build.config.src.clone())
+            .args(["--print", "target-list"])
+            .output()
+            .expect("Should have been able to fetch the target list.");
```

Comment:
This runs the default `rustc` from the operating system, it is wrong and may not even exist. Instead, you should use the `rustc` from stage0/beta.

Here is a reference from bootstrap sources:

https://github.com/rust-lang/rust/blob/c25473ff62a99541426423e8ef41c63d71e0a4a0/src/bootstrap/src/lib.rs#L347